pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -698,45 +756,56 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1beta1.Pip
 	}
 
 	for _, rpt := range nextRpts {
+		if rpt.IsFinalTask(pipelineRunFacts) {
+			c.setFinallyStartedTimeIfNeeded(pr)
```

Comment:
should this function also receive a pointer to pipelinerun facts and be responsible for updating that as well?

Reply:
Good idea!